ops/services/alerts/app_service_metrics/exceptions.tf

```diff
@@ -4,15 +4,78 @@ resource "azurerm_monitor_diagnostic_setting" "collect_appserviceconsolelogs" {
   target_resource_id         = var.app_service_id
   log_analytics_workspace_id = data.azurerm_log_analytics_workspace.global.id
 
+  log {
+    category = "AppServiceAntivirusScanAuditLogs"
+    enabled  = false
+    retention_policy {
+      days    = 0
+      enabled = false
+    }
+  }
   log {
     category = "AppServiceConsoleLogs"
     enabled  = true
-
     retention_policy {
       days    = 7
       enabled = true
     }
   }
+  log {
+    category = "AppServiceFileAuditLogs"
+    enabled  = false
+    retention_policy {
+      days    = 0
+      enabled = false
+    }
+  }
+  log {
+    category = "AppServiceHTTPLogs"
+    enabled  = false
+    retention_policy {
+      days    = 0
+      enabled = false
+    }
+  }
+  log {
+    category = "AppServiceIPSecAuditLogs"
+    enabled  = false
+    retention_policy {
+      days    = 0
+      enabled = false
+    }
+  }
+  log {
+    category = "AppServicePlatformLogs"
+    enabled  = false
+    retention_policy {
+      days    = 0
+      enabled = false
+    }
+  }
+  log {
+    category = "AppServiceAppLogs"
```

Comment:
I'll look at this again; this was added to reduce churn on this resource and make the `terraform plan` output generally easier to read.

At this point, I need to remember in what way it was churning, though. :smile: I'll do so by running some tf plans tomorrow. 

That said, and to be clear, this isn't vital to the upgrade, so we can cut it if we end up on the fence about it.

Reply:
After looking into this, I've rediscovered that I added these log/metric blocks because it cuts down on the resource's terraform output and churn/updates. Without the explicit log/metric blocks, terraform sets all the fields to null. After our updates go through, Azure will add them back with enabled set to false. Our terraform plan looks like this:

```
  # module.metric_alerts.azurerm_monitor_diagnostic_setting.collect_appserviceconsolelogs will be updated in-place
  ~ resource "azurerm_monitor_diagnostic_setting" "collect_appserviceconsolelogs" {
        id                         = "/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/prime-simple-report-dev/providers/Microsoft.Web/sites/simple-report-api-dev2|Dev2 API App Service console logs"
        name                       = "Dev2 API App Service console logs"
        # (2 unchanged attributes hidden)

      - log {
          - category = "AppServiceAntivirusScanAuditLogs" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }
      - log {
          - category = "AppServiceAppLogs" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }
      - log {
          - category = "AppServiceAuditLogs" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }
      - log {
          - category = "AppServiceConsoleLogs" -> null
          - enabled  = true -> null

          - retention_policy {
              - days    = 7 -> null
              - enabled = true -> null
            }
        }
      + log {
          + category = "AppServiceConsoleLogs"
          + enabled  = true

          + retention_policy {
              + days    = 7
              + enabled = true
            }
        }
      - log {
          - category = "AppServiceFileAuditLogs" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }
      - log {
          - category = "AppServiceHTTPLogs" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }
      - log {
          - category = "AppServiceIPSecAuditLogs" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }
      - log {
          - category = "AppServicePlatformLogs" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }

      - metric {
          - category = "AllMetrics" -> null
          - enabled  = false -> null

          - retention_policy {
              - days    = 0 -> null
              - enabled = false -> null
            }
        }
    }
```

With the explicit log/metric blocks, our terraform output for this resource looks like this:
```
pssst, pretend no text is here because nothing on this resource has changed. - Alis
```

So, while this may seem like we're disabling a bunch of stuff, in reality, we're just matching our terraform configuration up with reality.